tests/wpt_tests.cpp

```diff
@@ -172,7 +187,7 @@ bool setters_tests_encoding() {
 
         // TODO: Handle invalid utf-8 tests too.
         if (!element["expected"]["hostname"].get(expected)) {
-          ada::set_host(base, std::string{new_value});
+          ada::set_host(base, std::string{new_value}, type);
```

Comment:
I think encoding type should not be a parameter and it should use base’s encoding type.

Reply:
It is part of the API right now. 

https://github.com/ada-url/ada/blob/f0298674750eeae848c5859aff062e91ef248e4c/include/ada/implementation.h#L67


Either the parameter should not appear in the public-facing API or else we must test it.